### PR TITLE
feat(build): support to build on MacOS

### PR DESCRIPTION
**Pull request by acelyc111**
ref issue: https://github.com/apache/incubator-pegasus/issues/813

### CODE REVIEW

File: src/server/hashkey_transform.h [hunk 1]
```diff
@@ -46,7 +47,7 @@ class HashkeyTransform : public rocksdb::SliceTransform
         }
 
         // hash_key_len is in big endian
-        uint16_t hash_key_len = be16toh(*(int16_t *)(src.data()));
+        uint16_t hash_key_len = ::dsn::endian::ntoh(*(uint16_t *)(src.data()));
```
Comment (levy5307) [inline src/server/hashkey_transform.h:50]: ::dsn::endian::ntoh --> dsn::endian::ntoh

**Comment by Smityz** [comment]
Great job, after this PR merged, could you update the document on this page?
https://pegasus.apache.org/docs/build/compile-from-source/

**Comment by Smityz** [comment]
Is it possible to add CI step for MACOS?

**Comment by acelyc111** [comment]
> Is it possible to add CI step for MACOS?

We can do it in  the next step, similar to how rocksdb do, https://github.com/facebook/rocksdb/blob/7711f8cbb42dd96111a133099b029b6fabff6f48/.circleci/config.yml#L167

**Review by levy5307 — APPROVED** [review]

**Review by Shuo-Jia — APPROVED** [review]

**Pull request merged**